app/src/main/java/co/condorlabs/emerald/components/textfield/EmeraldTextFieldPassword.kt

```diff
@@ -0,0 +1,94 @@
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.LocalTextStyle
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.TextFieldColors
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import co.condorlabs.emerald.components.utils.Empty
+
+@Composable
+fun EmeraldTextFieldPassword(
+    state: EmeraldTextFieldState,
+    onValueChange: (String) -> Unit,
+    label: String,
+    modifier: Modifier = Modifier,
+    placeholder: String = Empty,
+    helperTextStart: String = Empty,
+    helperTextEnd: String = Empty,
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    singleLine: Boolean = true,
+    maxLines: Int = MaxLinesTextFieldDefault,
+    keyboardActions: KeyboardActions = KeyboardActions(),
+    textStyle: TextStyle = LocalTextStyle.current.copy(fontWeight = FontWeight.Normal),
+    leadingIcon: @Composable (() -> Unit)? = null,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    shape: Shape = MaterialTheme.shapes.small,
+    colors: TextFieldColors = emeraldTextFieldColors(),
+) {
+    val passwordVisible = rememberSaveable { mutableStateOf(false) }
+
+    EmeraldTextField(
+        state = state,
+        onValueChange = onValueChange,
+        label = label,
+        modifier = modifier,
+        placeholder = placeholder,
+        helperTextStart = helperTextStart,
+        helperTextEnd = helperTextEnd,
+        enabled = enabled,
+        readOnly = readOnly,
+        singleLine = singleLine,
+        maxLines = maxLines,
+        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+        keyboardActions = keyboardActions,
+        textStyle = textStyle,
+        leadingIcon = leadingIcon,
+        trailingIcon = {
+            val image = if (passwordVisible.value) {
+                Icons.Filled.Visibility
+            } else {
+                Icons.Filled.VisibilityOff
+            }
+
+            val description = if (passwordVisible.value) "Hide password" else "Show password"
```

Comment:
these strings should be in the string resources

Reply:
Done